proc_parent.go

```diff
@@ -156,7 +156,9 @@ func (mp *parent) sendSignal(s os.Signal) {
 	if mp.childCmd != nil && mp.childCmd.Process != nil {
 		if err := mp.childCmd.Process.Signal(s); err != nil {
 			mp.debugf("signal (%s) failed (%s), assuming child process died unexpectedly", s, err)
-			if !mp.Supervise {
+			//if we receive a SIGURG during shutdown
+			//don't exit with an error code
+			if !mp.Supervise && s.String() != "urgent I/O condition" {
```

Comment:
Does comparing the signal directly to [the SIGURG constant](https://pkg.go.dev/syscall#SIGURG) work? Then you wouldn't have to string-match.

Reply:
Turns out it does. Though reliably sending that signal to the program was a bit challenging.